tests/pyfunc/test_model_export_with_loader_module_and_data_path.py

```diff
@@ -182,34 +182,55 @@ def predict(pdf):
     assert res.dtypes.to_dict() == expected_types
     pdf["b"] = pdf["b"].astype(np.int64)
 
-    # 3. double -> float raises
+    # 3. unsigned int -> long works
+    pdf["b"] = pdf["b"].astype(np.uint32)
+    res = pyfunc_model.predict(pdf)
+    assert all((res == pdf[input_schema.column_names()]).all())
+    assert res.dtypes.to_dict() == expected_types
+    pdf["b"] = pdf["b"].astype(np.int64)
+
+    # 4. unsigned int -> int raises
+    pdf["a"] = pdf["a"].astype(np.uint32)
+    with pytest.raises(MlflowException) as ex:
+        pyfunc_model.predict(pdf)
+    assert "Incompatible input types" in str(ex)
+    pdf["a"] = pdf["a"].astype(np.int32)
+
+    # 5. double -> float raises
     pdf["c"] = pdf["c"].astype(np.float64)
     with pytest.raises(MlflowException) as ex:
         pyfunc_model.predict(pdf)
     assert "Incompatible input types" in str(ex)
     pdf["c"] = pdf["c"].astype(np.float32)
 
-    # 4. float -> double works
+    # 6. float -> double works
     pdf["d"] = pdf["d"].astype(np.float32)
     res = pyfunc_model.predict(pdf)
     assert res.dtypes.to_dict() == expected_types
     assert "Incompatible input types" in str(ex)
     pdf["d"] = pdf["d"].astype(np.int64)
 
-    # 5. floats -> ints raises
+    # 7. int -> float raises
     pdf["c"] = pdf["c"].astype(np.int32)
     with pytest.raises(MlflowException) as ex:
         pyfunc_model.predict(pdf)
     assert "Incompatible input types" in str(ex)
     pdf["c"] = pdf["c"].astype(np.float32)
 
+    # 8. int -> double works
+    pdf["d"] = pdf["d"].astype(np.int32)
+    pyfunc_model.predict(pdf)
+    assert all((res == pdf[input_schema.column_names()]).all())
+    assert res.dtypes.to_dict() == expected_types
+
+    # 9. long -> double raises
     pdf["d"] = pdf["d"].astype(np.int64)
     with pytest.raises(MlflowException) as ex:
         pyfunc_model.predict(pdf)
     assert "Incompatible input types" in str(ex)
     pdf["d"] = pdf["d"].astype(np.float64)
 
-    # 6. ints -> floats raises
+    # 10. any float -> any int raises
```

Comment:
For completeness, do we need to test `double -> long` and `double -> float`?

Reply:
We might as well